fractal.sdk/src/main/java/com/yanchware/fractal/sdk/domain/environment/EnvironmentAggregate.java

```diff
@@ -66,6 +65,58 @@ public void createOrUpdate() throws InstantiatorException {
     }
   }
 
+  public void manageSecrets() throws InstantiatorException {
+    manageEnvironmentSecrets(managementEnvironment.getId(), managementEnvironment.getSecrets());
+
+    for (var environment : managementEnvironment.getOperationalEnvironments()) {
+      manageEnvironmentSecrets(environment.getId(), environment.getSecrets());
+    }
+  }
+
+  private void manageEnvironmentSecrets(EnvironmentIdValue environmentId, Collection<Secret> environmentSecrets) throws InstantiatorException {
+    var existingSecrets = fetchExistingSecrets(environmentId);
```

Comment:
You need here a list of secrets as a non-cloud-agent. I think you need to allow retrieval of secrets in batch from the service, just ensure you do not add values.